eval/api_runner.py

```diff
@@ -85,7 +106,31 @@ def process_row(row, api_url: str, api_type: str, num_beams: int, decimal_points
         else:
             generated_query = generated_query.strip()
 
+    if "logprobs" in r.json():
+        logprobs = r.json()["logprobs"]
+
     row["generated_query"] = generated_query
+    logprobs_display = []
+    for item in logprobs:
+        probs = list(item.values())
+        probs_to_append = {}
+        for prob in probs:
+            rank = prob["rank"]
+            logprob = prob["logprob"]
+            token = prob["decoded_token"]
+            probs_to_append.update(
+                {
+                    f"rank_{rank}_token": token,
+                    f"rank_{rank}_logprob": logprob,
+                    f"rank_{rank}_prob": 10**logprob,
+                }
+            )
+
+        probs_to_append["prob_diff"] = (
```

Comment:
just curious are we guaranteed to always have at least 2 prob's with rank 1 and 2? otherwise might be good to append the `prob_diff` more defensively (eg test for empty/nulls before retrieving/appending).

Reply:
Yup yup always guaranteed to have this since the `logprobs` parameter is hardcoded to 2!